pkg/naming/ports.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package naming
+
+import (
+	"strings"
+	"unicode/utf8"
+)
+
+// PortName returns a dns-safe string for the given name.
+// Any char that is not [a-z0-9] is replaced by "-" or "a".
+// Replacement character "a" is used only at the beginning or at the end of the name.
+// The function does not change length of the string.
+func PortName(name string) string {
+	var d []rune
+
+	for i, x := range strings.ToLower(name) {
+		if regex.Match([]byte(string(x))) {
+			d = append(d, x)
+		} else {
+			if i == 0 || i == utf8.RuneCountInString(name)-1 {
+				d = append(d, 'a')
+			} else {
+				d = append(d, '-')
+			}
+		}
+	}
+
+	return string(d)
+}
```

Comment:
Or using string replace instead of a loop with nested if/else?

```suggestion
func PortName(name string) string {
    // Mapping function to replace invalid characters with "-"
    mapper := func(r rune) rune {
        if unicode.IsLetter(r) || unicode.IsDigit(r) {
            return r
        }
        return '-'
    }

    // Map the name using the mapper function and replace "-" at the beginning or end with "a"
    result := strings.Map(mapper, name)
    if strings.HasPrefix(result, "-") {
        result = "a" + result[1:]
    }
    if strings.HasSuffix(result, "-") {
        result = result[:len(result)-1] + "a"
    }

    return result
}
```